docs/DeveloperGuide.md

```diff
@@ -1,18 +1,120 @@
 # Developer Guide
 
-## Design & implementation
+## Design
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Architecture
 
+The **Architecture Design** given above explains the high-level design of the App.
+Given below is a quick overview of each component. </br>
 
-## Product scope
-### Target user profile
+`Modman` is the main class of the application, and handles the app launch,
+initializing the appropriate classes to be used. </br>
+
+The rest of the app consists of four components:
+
+* Logic
+* UI
+* Storage
+* Data
+
+**How the four architecture components interact with each other and the ModMan class**
+
+The *Sequence diagram* below shows the interaction between components for when the user issues the command `add module CS2113T`
+
+![](uml/ArchitectureSequenceDiagram.png)
+
+The sections below give more details of each component.
+
+
+## Logic Component (Jianning)
+
+![logic](uml/ParserAndCommand.png)
```

Comment:
Perhaps multiplicity should be included in this and the following class diagrams?